x-pack/plugins/profiling/public/components/topn_functions.tsx

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useContext, useMemo } from 'react';
+
+import { EuiBasicTable, EuiHorizontalRule, EuiSpacer, EuiText } from '@elastic/eui';
+
+import { describeFrameType, StackFrameMetadata } from '../../common/profiling';
+import { FunctionContext } from './contexts/function';
+
+function getCalleeFunction(frame: StackFrameMetadata): string {
+  // In the best case scenario, we have the file names, source lines,
+  // and function names. However we need to deal with missing function or
+  // executable info.
+  const exeDisplayName = frame.ExeFileName ? frame.ExeFileName : describeFrameType(frame.FrameType);
+
+  // When there is no function name, only use the executable name
+  return frame.FunctionName ? exeDisplayName + ': ' + frame.FunctionName : exeDisplayName;
+}
+
+function getCalleeSource(frame: StackFrameMetadata): string {
```

Comment:
This looks like code that warrants a unit test, but up to you of course.

Reply:
Thanks for the suggestion. It reminded me that these two methods should live closer to the definition of the `StackFrameMetadata` interface, so I moved the methods while adding tests.